static/app/components/nav/config.tsx

```diff
@@ -0,0 +1,192 @@
+          {label: t('Crons'), to: `/${prefix}/crons/`},
+        ],
+      },
+      {
+        label: t('Insights'),
+        icon: <IconGraph />,
+        check: {features: 'insights-entry-points'},
+        submenu: [
+          {label: MODULE_SIDEBAR_TITLE, to: `/${prefix}/${moduleURLBuilder('http')}/`},
+          {label: MODULE_TITLES.db, to: `/${prefix}/${moduleURLBuilder('db')}/`},
+          {
+            label: MODULE_TITLES.resource,
+            to: `/${prefix}/${moduleURLBuilder('resource')}/`,
+          },
+          {
+            label: MODULE_TITLES.app_start,
+            to: `/${prefix}/${moduleURLBuilder('app_start')}/`,
+          },
+          {
+            label: MODULE_TITLES['mobile-screens'],
+            to: `/${prefix}/${moduleURLBuilder('mobile-screens')}/`,
+            check: {features: 'insights-mobile-screens-module'},
+          },
+          {label: MODULE_TITLES.vital, to: `/${prefix}/${moduleURLBuilder('vital')}/`},
+          {label: MODULE_TITLES.cache, to: `/${prefix}/${moduleURLBuilder('cache')}/`},
+          {label: MODULE_TITLES.queue, to: `/${prefix}/${moduleURLBuilder('queue')}/`},
+          {
+            label: MODULE_TITLES.ai,
+            to: `/${prefix}/${moduleURLBuilder('ai')}/`,
+            check: {features: 'insights-entry-points'},
+          },
+        ],
+      },
+      {
+        label: t('Perf.'),
+        to: '/performance/',
+        icon: <IconLightning />,
+        check: {features: 'performance-view', hook: 'performance-sidebar-item'},
+      },
+      {
+        label: t('Boards'),
+        to: '/dashboards/',
+        icon: <IconDashboard />,
+        check: {
+          features: ['discover', 'discover-query', 'dashboards-basic', 'dashboards-edit'],
+          hook: 'dashboards-sidebar-item',
+        },
+      },
+      {label: t('Alerts'), to: `/${prefix}/alerts/rules/`, icon: <IconSiren />},
+      NAV_DIVIDER,
+      // {label: t('Help'), to: '', icon: <IconQuestion />},
+      // {label: t('New'), to: '', icon: <IconBroadcast />},
+      // {label: t('Stats'), to: '', icon: <IconStats />},
+      {
+        label: t('Settings'),
+        to: `/settings/${organization.slug}/`,
+        icon: <IconSettings />,
+      },
+    ],
+    [organization, moduleURLBuilder, prefix]
+  );
+
+  const formatted = useMemo(() => formatNavItems({location}, items), [location, items]);
+  return formatted;
+}
+
+function formatNavItems(
+  context: {location: ReturnType<typeof useLocation>},
+  items: NavItemRaw[]
+): NavItemsResult {
+  const sidebar = items
+    .filter(item => !!item)
+    .map(item => (typeof item === 'object' ? resolveSidebarItem(item) : item));
+  const primary = splitAtDivider(sidebar);
+  const {submenu = []} = primary.body.find(
+    item => getActiveStatus(item, context.location) !== ActiveStatus.INACTIVE
+  ) ?? {
+    submenu: [],
+  };
+  const secondary = splitAtDivider(submenu);
+
+  return {
+    primary,
+    secondary,
+  };
+}
```

Comment:
Is there a way we could simplify this logic entirely and remove the NAV_DIVIDER as well? I think it would have a lot of nice downstream effects on simplifying the type definitions and rendering

Right now the the splitting procedure assumes that there is only a single divider, but that rule is not enforced, so someone could think that adding a divider inside the config would just render a divider, but it would actually result in a wrong split between the groups and not rendering some items (or attempt to render a symbol and crash)

Reply:
Great feedback—this has been restructured as part of [92d70b5](https://github.com/getsentry/sentry/pull/77665/commits/92d70b59ab5a518fa704b5483e958590d2fbe50c).